### PR TITLE
feat: add react-query and navigate to edit event type once created

### PR DESCRIPTION
**Pull request by mihaic195**
- When a user creates an event type, he should be sent to the Edit Event type page.
- Also added `react-query`. Creating a new event type is done through `react-query` in this PR.

### CODE REVIEW

File: lib/app-providers.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { createTelemetryClient, TelemetryProvider } from "@lib/telemetry";
+import { Provider } from "next-auth/client";
+import { QueryClient, QueryClientProvider } from "react-query";
+import { Hydrate } from "react-query/hydration";
+
+export const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false, // don't fetch when window is focused
```
Comment (KATT) [inline lib/app-providers.tsx:10]: why not? it's a nice feature

File: lib/app-providers.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      retry: false, // if it fails don't retry (you can configure this on a specific resource)
```
Comment (KATT) [inline lib/app-providers.tsx:11]: why not? it's a nice feature

File: lib/mutations/event-types/create-event-type.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const createEventType = async (body: any) => {
+  // TODO: Add validation
+  const response = await fetch("/api/availability/eventtype", {
+    method: "POST",
+    body: body,
```
Comment (KATT) [inline lib/mutations/event-types/create-event-type.ts:5]:
```suggestion
const createEventType = async (data: { /* strict types */ }) => {
  const response = await fetch("/api/availability/eventtype", {
    method: "POST",
    body: JSON.stringify(data),
```

File: pages/api/availability/eventtype.ts [hunk 4]
```diff
@@ -61,13 +62,13 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     };
 
     if (req.method == "POST") {
-      await prisma.eventType.create({
+      const event_type = await prisma.eventType.create({
```
Comment (emrysal) [inline pages/api/availability/eventtype.ts:65]: minor: eventType

Q: If we are going to add the created resource in the success 201 response shall we go all the way - remove the message entirely? (Maybe add a location header if possible)

Comment (mihaic195) [reply pages/api/availability/eventtype.ts:65]: IMHO a proper create endpoint is one which returns the created object. Same for update.

Comment (emrysal) [reply pages/api/availability/eventtype.ts:65]: Sorry I mean the literal `message` - when you return the full created object the success is kinda given; ipv:

```javascript
- res.status(201).json({ message: "Event created successfully", event_type });
+ res.status(201).json(eventType);
```

Comment (mihaic195) [reply pages/api/availability/eventtype.ts:65]: Ah, yes. I think it can be removed indeed.

File: pages/_app.tsx [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
-function MyApp({ Component, pageProps }: AppProps) {
+// Workaround for https://github.com/zeit/next.js/issues/8592
+export type AppProps = NextAppProps & {
+  /** Will be defined only is there was an error */
+  err?: Error;
+};
+
+function MyApp({ Component, pageProps, err }: AppProps) {
   return (
-    <TelemetryProvider value={createTelemetryClient()}>
```
Comment (PeerRich) [inline pages/_app.tsx:9]: where is Telementry now?

Comment (mihaic195) [reply pages/_app.tsx:9]: I've created an `app-providers.tsx` file with all of them, and then in the `_app.tsx` we only keep the `<AppProviders />`.

Comment (PeerRich) [reply pages/_app.tsx:9]: ah epic! much better solution

File: pages/_app.tsx [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
-      <Provider session={pageProps.session}>
-        <Head>
-          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-        </Head>
-        <Component {...pageProps} />
-      </Provider>
-    </TelemetryProvider>
+    <AppProviders>
```
Comment (PeerRich) [inline pages/_app.tsx:14]: great 👍

File: lib/app-providers.tsx (hunk 1 above)
Comment (PeerRich) [reply lib/app-providers.tsx:10]: i like this too. dont see downside of enabling it

**Pull request closed without merging**